src/backend/XmlHttpRequest.ts

```diff
@@ -31,6 +33,10 @@ export interface XmlHttpRequestOptions {
   // Used as the URL prefix for fetched files.
   // Default: Fetch files relative to `url`.
   baseUrl?: string;
+  // Whether to use Fetch or XmlHttpRequest.
+  // Can be 'xhr', 'fetch', or 'auto'.
+  // Default: 'auto' which uses fetch if its available, and XHR as a fallback
```

Comment:
Oh, sorry for the delay! I see what you mean now. In the Node environment, it's also plausible that someone would want to use the `HTTP` API.

Another road to go, and maybe this is crazy, is to make separate `XmlHttpRequest`, `FetchFS`, etc file systems. Abstract away the core `HttpRequestFS` into an abstract class, and extend it with the `downloadAsync` / `downloadSync` / `getSizeAsync` / etc methods. This would also easily preserve backwards compatibility, and make it possible to exclude environment-specific file systems from e.g. the browser bundle.